nntrainer/tensor/tensor.cpp

```diff
@@ -1228,6 +1232,26 @@ void Tensor::dropout_mask(float dropout) {
   }
 }
 
+Tensor Tensor::zoneout_mask(float zoneout) {
+  Tensor ret(getDim());
+  zoneout_mask(ret, zoneout);
+  return ret;
+}
+
+void Tensor::zoneout_mask(Tensor &opposite, float zoneout) {
+  opposite.setRandBernoulli(zoneout);
+  float *data = getData();
+  float *opposite_data = opposite.getData();
+
+  for (unsigned int i = 0; i < size(); ++i) {
+    if (opposite_data[i]) {
```

Comment:
important: we can't check float == 0, it's gotta be done like float  < 1e-6

Reply:
Done